src/plugins/discover/public/application/components/sidebar/discover_sidebar.tsx

```diff
@@ -98,12 +94,12 @@ export function DiscoverSidebar({
   const [showFields, setShowFields] = useState(false);
   const [fields, setFields] = useState<IndexPatternFieldList | null>(null);
   const [fieldFilterState, setFieldFilterState] = useState(getDefaultFieldFilter());
-  const services = getServices();
+  const services = useMemo(() => getServices(), []);
```

Comment:
FYI @sulemanof without `useMemo` the jest test was stuck and timed out

Reply:
This looks rather weird, because this means the services could change, but we won't get a fresh version?
@kertal

Reply:
@lizozom it's just the data service of services used here to return a fieldList if there's no indexPattern or fieldCounts given. Code seems to have changed since the last time I've reviewed / worked with it. but there should be no issue, it's just used to provide a fallback

Reply:
OK, the answer could be pretty simple if I can simply move `const services = getServices()` upper, outside of the React component. (services are always static from setup/start). But I can't do it now, because the component is used as ng-react directive, and angular compiles in setup method of discover plugin.
So the memoization is only needed to get rid of endless effect running in here https://github.com/elastic/kibana/pull/63473/files#diff-57f3e466091bbcf7660e7a7f6dbb3400L103 ;
I could get rid of passing down services into `getIndexPatternFieldList`, and call `getServices` right in that function, but I prefer get rid of `getServices` at all in scope of https://github.com/elastic/kibana/issues/63596